sklearn/model_selection/_split.py

```diff
@@ -50,8 +51,8 @@ class BaseCrossValidator(metaclass=ABCMeta):
 
     Implementations must define `_iter_test_masks` or `_iter_test_indices`.
     """
-
-    def split(self, X, y=None, groups=None):
+    @_deprecate_positional_args
+    def split(self, X, y=None, *, groups=None):
```

Comment:
Not sure how valuable it is to make `groups` kwonly in a method that currently always takes <=3 params

Reply:
My thinking was that very often the third argument after `X, y` is `sample_weight`, and this removes that ambiguity.

Reply:
On the other hand, it seems at least we internally heavily pass `groups` as a positional arg. So I guess that itself is a verdict that it's breaking too much code. I'll revert this one.